src/silx/utils/retry.py

```diff
@@ -127,6 +129,59 @@ def wrapper(*args, **kw):
     return decorator
 
 
+def retry_iterator(
```

Comment:
From what I tested, it seems possible to add support for iterator to `retry` instead of having a second one:

```python
def printer(prefix=''): 
     def d(f): 
         if inspect.isgeneratorfunction(f): 
             @wraps(f) 
             def w(*args, **kwargs): 
                 print(f"{prefix} generatorfunction {args}, {kwargs}") 
                 for r in f(*args, **kwargs): 
                     print('yield', r) 
                     yield r 
             return w 
         @wraps(f) 
         def w(*args, **kwargs): 
             print(f"{prefix} function {args}, {kwargs}") 
             return f(*args, **kwargs) 
         return w 
     return d
```

```python
In [22]: @printer("test") 
    ...: def f(): 
    ...:     return 1 
    ...:                                                                        

In [23]: f()                                                                    
test function (), {}
Out[23]: 1

In [25]: @printer("test") 
    ...: def g(): 
    ...:     yield 1 
    ...:                                                         

In [27]: for i in g(): 
    ...:     print(i) 
    ...:                                                                        
test generatorfunction (), {}
yield 1
1
```

Reply:
Thanks for the suggestion.